tests/qnn/test_keras.py

```diff
@@ -75,6 +98,14 @@ def model_dm(get_circuit_dm, n_qubits, output_dim):
     return m
 
 
+@pytest.fixture(autouse=True)
+def reset_keraslayer_input_arg():
```

Comment:
How does this work?

Reply:
So this and the other place where `finally:` appeared are both due to the fact that `KerasLayer` has a method to modify its own class property `input_arg` (still don't understand why we need such method and testing for them though) so I would like to safeguard no other tests launched together are affected.